### ISSUE TITLE
chapter 3. shape deformation  ARAP Demo

### ISSUE THREAD

**Issue by sc0m3r** [libigl/libigl-python-bindings#43]
The ARAP demo code here, https://libigl.github.io/libigl-python-bindings/tutorials/
chapter 3. shape deformation. 

the demo code yields a poor/unexpected result (attached image)

I went through the the older py_igl implementation to see if I was missing something, but nothing turned up.  Any help would be greatly appreciated. I also tried the tutorial with a different mesh. 

Thanks,

<img width="967" alt="Screen Shot 2020-04-02 at 1 47 43 PM" src="https://user-images.githubusercontent.com/33268556/78310508-0bb53c80-7503-11ea-9ac0-d42c0086df3d.png">

**Comment by teseoch** [comment]
are you using the latest version of all libs?

**Comment by sc0m3r** [comment]
I installed igl through Conda. Here is the version(s) that I have
installed.

Thanks,

Igl verison:

igl                       0.4.1            py36h8bd51ef_0    conda-forge

Full anaconda list :

# Name                    Version                   Build  Channel

appnope                   0.1.0            py36hf537a9a_0

apptools                  4.4.0                    py36_1    anaconda

attrs                     19.3.0                     py_0

backcall                  0.1.0                    py36_0

bleach                    3.1.0                    py36_0

bzip2                     1.0.8                h1de35cc_0    anaconda

ca-certificates           2020.1.1                      0    anaconda

certifi                   2019.11.28               py36_1    anaconda

configobj                 5.0.6                    py36_1    anaconda

curl                      7.69.1               ha441bb4_0    anaconda

cycler                    0.10.0                     py_2    conda-forge

dbus                      1.13.12              h90a0687_0

decorator                 4.4.2                      py_0

defusedxml                0.6.0                      py_0

entrypoints               0.3                      py36_0

envisage                  4.8.0                      py_0    anaconda

expat                     2.2.6                h0a44026_0

freetype                  2.10.1               h8da9a1a_0    conda-forge

future                    0.18.2                   py36_0    anaconda

gettext                   0.19.8.1             h15daf44_3

glib                      2.63.1               hd977a24_0

gmp                       6.2.0                h4a8c4bd_2    conda-forge

h5py                      2.10.0          nompi_py36h106b333_102
conda-forge

hdf4                      4.2.13               h39711bb_2    anaconda

hdf5                      1.10.5          nompi_h3e39495_1104    conda-forge

icu                       58.2                 h4b95b61_1

igl                       0.4.1            py36h8bd51ef_0    conda-forge

importlib_metadata        1.5.0                    py36_0

ipydatawidgets            4.0.1                      py_0    conda-forge

ipykernel                 5.1.4            py36h39e3cac_0

ipyscales                 0.4.0                      py_1    conda-forge

ipython                   7.13.0           py36h5ca1d4c_0

ipython_genutils          0.2.0                    py36_0

ipywidgets                7.5.1                      py_0

jedi                      0.16.0                   py36_1

jinja2                    2.11.1                     py_0

jpeg                      9c                h1de35cc_1001    conda-forge

jsoncpp                   1.8.4                h04f5b5a_0    anaconda

jsonschema                3.2.0                    py36_0

jupyter                   1.0.0                    py36_7

jupyter_client            6.1.0                      py_0

jupyter_console           6.1.0                      py_0

jupyter_core              4.6.1                    py36_0

kiwisolver                1.1.0            py36h863e41a_1    conda-forge

krb5                      1.17.1               hddcf347_0    anaconda

libblas                   3.8.0               16_openblas    conda-forge

libcblas                  3.8.0               16_openblas    conda-forge

libcurl                   7.69.1               h051b688_0    anaconda

libcxx                    9.0.1                         1    conda-forge

libcxxabi                 4.0.1                hcfea43d_1

libedit                   3.1.20181209         hb402a30_0

libffi                    3.2.1                h475c297_4

libgfortran               4.0.0                         2    conda-forge

libiconv                  1.15                 hdd342a3_7

liblapack                 3.8.0               16_openblas    conda-forge

libnetcdf                 4.6.2             h1a02027_1003    conda-forge

libopenblas               0.3.9                h3d69b6c_0    conda-forge

libpng                    1.6.37               ha441bb4_0

libsodium                 1.0.16               h3efe00b_0

libssh2                   1.9.0                ha12b0ac_1    anaconda

libtiff                   4.1.0                hcb84e12_0    anaconda

libxml2                   2.9.9                hf6e021a_1    anaconda

llvm-openmp               9.0.1                h28b9765_2    conda-forge

lz4-c                     1.8.3             h6de7cb9_1001    conda-forge

markupsafe                1.1.1            py36h1de35cc_0

matplotlib                3.2.1                         0    conda-forge

matplotlib-base           3.2.1            py36h83d3ec1_0    conda-forge

mayavi                    4.7.1            py36ha91d4f2_2    conda-forge

meshplot                  v0.3.2                     py_0    conda-forge

metis                     5.1.0             h4a8c4bd_1005    conda-forge

mistune                   0.8.4            py36h1de35cc_0

nb_conda                  2.2.1                    py36_2    conda-forge

nb_conda_kernels          2.2.3                    py36_0    conda-forge

nbconvert                 5.6.1                    py36_0

nbformat                  5.0.4                      py_0

ncurses                   6.2                  h0a44026_0

notebook                  6.0.3                    py36_0

numpy                     1.18.1           py36hdc5ca10_1    conda-forge

openssl                   1.1.1                h1de35cc_0    anaconda

pandoc                    2.2.3.2                       0

pandocfilters             1.4.2                    py36_1

parso                     0.6.2                      py_0

pcre                      8.43                 h0a44026_0

pexpect                   4.8.0                    py36_0

pickleshare               0.7.5                    py36_0

pip                       20.0.2                   py36_1

polyfempy                 0.5.2            py36h6345f57_4    conda-forge

prometheus_client         0.7.1                      py_0

prompt-toolkit            3.0.4                      py_0

prompt_toolkit            3.0.4                         0

ptyprocess                0.6.0                    py36_0

pyface                    6.1.2                    py36_0    anaconda

pygments                  2.6.1                      py_0

pyparsing                 2.4.6                      py_0    conda-forge

pyqt                      5.9.2            py36h655552a_2

pyrsistent                0.15.7           py36h1de35cc_0

python                    3.6.10               hc70fcce_1

python-dateutil           2.8.1                      py_0

python_abi                3.6                     1_cp36m    conda-forge

pythreejs                 2.2.0              pyh8c360ce_0    conda-forge

pyzmq                     18.1.1           py36h0a44026_0

qt                        5.9.7                h468cd18_1

qtconsole                 4.7.2                      py_0

qtpy                      1.9.0                      py_0

readline                  8.0                  h1de35cc_0

scipy                     1.4.1            py36h82752d6_0    conda-forge

send2trash                1.5.0                    py36_0

setuptools                46.1.1                   py36_0

sip                       4.19.8           py36h0a44026_0

six                       1.14.0                   py36_0

sqlite                    3.31.1               ha441bb4_0

suitesparse               5.6.0                h0e59142_0    conda-forge

svgwrite                  1.3.1                      py_0    conda-forge

tbb                       2019.9               ha1b3eb9_1    conda-forge

terminado                 0.8.3                    py36_0

testpath                  0.4.4                      py_0

tk                        8.6.8                ha441bb4_0

tornado                   6.0.4            py36h1de35cc_1

traitlets                 4.3.3                    py36_0

traits                    5.2.0            py36h1de35cc_0    anaconda

traitsui                  6.1.3                      py_0    anaconda

traittypes                0.2.1                      py_1    conda-forge

vtk                       8.2.0           py36hcba847f_203    conda-forge

wcwidth                   0.1.8                      py_0

webencodings              0.5.1                    py36_1

wheel                     0.34.2                   py36_0

widgetsnbextension        3.5.1                    py36_0

wildmeshing               0.3.0.2          py36h04c165b_0    conda-forge

xz                        5.2.4                h1de35cc_4

zeromq                    4.3.1                h0a44026_3

zipp                      2.2.0                      py_0

zlib                      1.2.11               h1de35cc_3

zstd                      1.3.7                h5bba6e5_0    anacond
                0.4.1            py36h8bd51ef_0    conda-forge

**Comment by ilkedemir** [comment]
I observe the same problem with the current mesh and a different mesh. 

igl version: 0.3.1
meshplot version: 0.3.2

**Comment by teseoch** [comment]
strange, and you are just running the default code.
Which os are you using?

**Comment by ilkedemir** [comment]
Yes, running the default tutorial in Jupyter notebook (6.0.3) on mac (10.15.1).

**Comment by ilkedemir** [comment]
The vertex indices for faces are broken in `meshplot`. I haven't dug in detail, but a quick fix for the sake of this tutorial is to change [the following lines](https://github.com/skoch9/meshplot/blob/master/meshplot/Viewer.py#L216) in `meshplot`:

```
        else:
            f = f.astype("uint32", copy=False).ravel()
            ba_dict["index"] = p3s.BufferAttribute(f, normalized=False)
```
to

```
        #else:
        f = f.astype("uint32", copy=False).ravel()
        ba_dict["index"] = p3s.BufferAttribute(f, normalized=False)
```
in order to preserve the face associations. The coloring changes to vertex coloring, but at least you'll be able to see the mesh deformation. I would also enter an issue to `meshplot` repo.

**Comment by sc0m3r** [comment]
Thanks,

That works for me.

**Comment by skoch9** [comment]
This was indeed a problem in meshplot and is fixed now with version 0.4.0.
